admin_manual/configuration/server/security/oauth2.rst

```diff
@@ -0,0 +1,174 @@
+     |         |<---(E)----- Access Token -------------------'
+     +---------+       (w/ Optional Refresh Token)
+
+
+The OAuth2 App
+--------------
+
+OAuth2 support is available in ownCloud via `an OAuth2 application`_ which is available from the ownCloud Marketplace.
+The app aims to:
+
+#. Connect ownCloud clients (both desktop and mobile) in a standardized and secure way.
+#. Make 3rd party software integrations easier by providing an unified authorization interface.
+
+Endpoints
+~~~~~~~~~
+
+================= =======================================
+Description       URI
+================= =======================================
+Authorization URL ``/index.php/apps/oauth2/authorize``
+Access Token URL  ``/index.php/apps/oauth2/api/v1/token``
+================= =======================================
+
+Protocol Flow
+~~~~~~~~~~~~~
+
+Client Registration
+^^^^^^^^^^^^^^^^^^^^
+
+The clients first have to be registered in the admin settings: ``/settings/admin?sectionid=authentication``.
+You need to specify a name for the client (the name is unrelated to the OAuth 2.0 protocol and is just used to recognize it later) and the redirection URI.
+A client identifier and client secret are generated when adding a new client, which both consist of 64 characters.
+For further information about client registration, please refer to `the official client registration RFC from the IETF`_.
+
+Authorization Request
+^^^^^^^^^^^^^^^^^^^^^
+
+For every registered client an authorization request can be made.
+The client redirects the resource owner to the authorization URL and requests authorization.
+The following URL parameters have to be specified:
+
+================= ========== ========================================================================================
+Parameter         Required   Description
+================= ========== ========================================================================================
+``response_type``  yes       Needs to be `code` because at this time only the authorization code flow is implemented.
+``client_id``      yes       The client identifier obtained when registering the client.
+``redirect_uri``   yes       The redirection URI specified when registering the client.
+``state``          no        Can be set by the client "to maintain state between the request and callback".
+                             See `RFC 6749`_ for more information.
+================= ========== ========================================================================================
+
+For further information about client registration, please refer to `the official authorization request RFC from the IETF`_.
+
+Authorization Response
+^^^^^^^^^^^^^^^^^^^^^^^
+
+After the resource owner's authorization, the app redirects to the `redirect_uri` specified in the authorization request and adds the authorization code as URL parameter `code`.
+An authorization code is valid for 10 minutes.
```

Comment:
What happens after the 10min? Is there a user interaction necessary or are there processes running in the background automatically ? If yes which or what happens?

Reply:
Isn't this rather something for a developer documentation? I think here we need to document what administrators need to know. Technical overview, benefits, configuration details, minimum client versions, etc..

Reply:
I agree with you @pmaier1. @mmattel, without having looked over the code, I'd suggest that on the next request, the user would have to grant access again, unless the ability to refresh the token is available.

Reply:
Actually, @pmaier1, on reading the text again, it makes sense to keep the background information and configuration documentation together, rather than have the admin/user search for one or the other. I'd suggest to keep it as it is.

Reply:
Proposal:
``An authorization code is valid for 10 minutes.`` --> ``For security reasons, an authorization code is valid for 10 minutes. If the client tries to use the authorization code more than once, the authorization server denies the request.``